### PR TITLE
Bjorncs/json feeder

### PR DESCRIPTION
**Pull request by bjorncs**
I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: vespa-feed-client/src/main/java/ai/vespa/feed/client/JsonFeeder.java [hunk 1]
```diff
@@ -284,6 +309,21 @@ String getDocumentJson(long start, long end) {
...
+
+        private final byte[] json;
+
+        SingleOperationParserAndExecutor(byte[] json) throws IOException {
+            super(factory.createParser(json), false);
+            this.json = json;
+        }
+
+        @Override
+        String getDocumentJson(long start, long end) {
```
Comment (jonmv) [inline vespa-feed-client/src/main/java/ai/vespa/feed/client/JsonFeeder.java:322]: We should probably just use `byte[]` as data carrier, all the way through to the HTTP layer?

Comment (jonmv) [reply vespa-feed-client/src/main/java/ai/vespa/feed/client/JsonFeeder.java:322]: Not very related to this PR, though :)

**Review by jonmv — APPROVED** [review]
Nice!

**Comment by bjorncs** [comment]
Rebased with master to resolve merge conflict. The 3 last commits are new.

**Review by jonmv — APPROVED** [review]

**Pull request merged**